repos/spack_repo/builtin/packages/hipblaslt/package.py

```diff
@@ -148,7 +165,12 @@ def setup_build_environment(self, env: EnvironmentModifications) -> None:
             env.set("ROCM_LD_PATH", f"{self.spec['llvm-amdgpu'].prefix}/bin/ld.lld")
             env.set("CC", f"{self.spec['llvm-amdgpu'].prefix}/bin/amdclang")
 
+        if self.spec.satisfies("@7.1") and self.run_tests:
+            env.append_flags("LDFLAGS", "-lstdc++fs")
+
     def patch(self):
+        py_ver = self.spec["python"].version[:-1]
```

Comment:
Python does have a 3-component version but if someone builds a python package that somehow doesn't have one (e.g. maybe they'll add a suffix, or `.dev0`,  or something) this will fail. Use `up_to()` to guarantee 2 components:

```suggestion
        py_ver = self.spec["python"].version.up_to(2)
```